### PR TITLE
Add ES 7.17.x to Supported Environment

### LINKED ISSUE #1808: Clarify supported versions of Elasticsearch
The supported versions of Elasticsearch on [this page](https://docs.camunda.io/docs/reference/supported-environments/#camunda-platform-8-self-managed) need to be updated. At the moment it shows 7.16.x, but we've had confirmations internally about some versions of Elasticsearch 7.17. This may need to be checked with the different component teams. 

* [Support ticket](https://jira.camunda.com/browse/SUPPORT-15365) 
* [Slack thread 1](https://camunda.slack.com/archives/C03UR0V2R2M/p1677676320682899)
* [Slack thread 2](https://camunda.slack.com/archives/C03UR0V2R2M/p1678371302974029)

### PR DESCRIPTION
**Pull request by akeller**
## What is the purpose of the change

Closes #1808. 

## Are there related marketing activities

No, but helpful for support.

## When should this change go live?

When ready, again to help clarify for support.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
Still waiting on Operate and Tasklist to confirm before moving this out of draft.

File: docs/reference/supported-environments.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Requirements for the components can be seen below:
 
 | Component                | Java version | Other requirements                                                                               |
 | ------------------------ | ------------ | ------------------------------------------------------------------------------------------------ |
-| Zeebe Broker and Gateway | OpenJDK 17+  | Elasticsearch 7.16.x (only if Elastic exporter is used)                                          |
+| Zeebe Broker and Gateway | OpenJDK 17+  | Elasticsearch 7.16.x, 7.17.x (only if Elastic exporter is used)                                  |
```
Comment (remcowesterhoud) [inline docs/reference/supported-environments.md:37]:
```suggestion
| Zeebe | OpenJDK 17+  | Elasticsearch 7.16.x, 7.17.x (only if Elastic exporter is used)                                  |
```

I know this already said Zeebe Broker and Gateway, but this is a bit strange. The gateway has nothing to do with Elasicsearch at all. Adding broker seems lik too much of an implementation detail to me. It's better to just refer to the component as `Zeebe`.

File: versioned_docs/version-8.1/reference/supported-environments.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Requirements for the components can be seen below:
 
 | Component                | Java version | Other requirements                                                                               |
 | ------------------------ | ------------ | ------------------------------------------------------------------------------------------------ |
-| Zeebe Broker and Gateway | OpenJDK 17+  | Elasticsearch 7.16.x (only if Elastic exporter is used)                                          |
+| Zeebe Broker and Gateway | OpenJDK 17+  | Elasticsearch 7.16.x, 7.17.x (only if Elastic exporter is used)                                  |
```
Comment (remcowesterhoud) [inline versioned_docs/version-8.1/reference/supported-environments.md:37]:
```suggestion
| Zeebe Broker and Gateway | OpenJDK 17+  | Elasticsearch 7.16.x, 7.17.x (only if Elastic exporter is used)                                  |
```

I know this already said Zeebe Broker and Gateway, but this is a bit strange. The gateway has nothing to do with Elasicsearch at all. Adding broker seems lik too much of an implementation detail to me. It's better to just refer to the component as `Zeebe`.

**Review by remcowesterhoud — APPROVED** [review]
Thanks @akeller, the versioning looks good from a Zeebe perspective. I do have a minor suggestions to change the Component to `Zeebe`

**Comment by akeller** [comment]
Grr... merge conflicts. Sorry for all the thrash here.

**Review by sdorokhova — APPROVED** [review]
Operate and Tasklist part looks good!

**Pull request merged**